applicationset/generators/generator_spec_processor.go

```diff
@@ -81,3 +91,33 @@ func mergeGeneratorTemplate(g Generator, requestedGenerator *argoprojiov1alpha1.
 
 	return *dest, err
 }
+
+// Currently for Matrix Generator. Allows interpolating the matrix's 2nd child generator with values from the 1st child generator
+// "params" parameter here are params from the 1st child generator. Uses similar process for interpreting values as the actual Application Template
+func interpolateGenerator(requestedGenerator *argoprojiov1alpha1.ApplicationSetGenerator, params []map[string]string) error {
+	tmplBytes, err := json.Marshal(requestedGenerator)
+	if err != nil {
+		log.WithError(err).WithField("requestedGenerator", requestedGenerator).Error("error marshalling requested generator for interpolation")
+		return err
+	}
+	tmpParams := make(map[string]string)
+	for _, currParam := range params {
+		for k, v := range currParam {
+			tmpParams[k] = v
+		}
+	}
```

Comment:
After this block of code, won’t `tmpParams` contain whatever was in the last map in the `params` list for a given key? Meaning that the templating below would always interpolate the last value for a given key instead of _each_ value for a given key?

If I’m understanding the use case, it seems `params` should be a map (instead of a list of maps) and then the matrix generator code would look something like:

```go
g0, err := m.getParams(appSetGenerator.Matrix.Generators[0], appSet, map[string]string{})

for _, a := range g0 {
  g1, err := m.getParams(appSetGenerator.Matrix.Generators[1], appSet, a)

  for _, b := range g1 {
    val, err := utils.CombineStringMaps(a, b)
    if err != nil {
      return nil, err
    }
    res = append(res, val)
  }
}
```

…where `g1` is determined for each param map `a` from `g0`.

Reply:
So this is how the "params" function parameter value looks like 
<img width="1273" alt="image" src="https://user-images.githubusercontent.com/29113963/163661670-ebd236bf-cb95-41c0-8ce6-ad2f92f7d0c1.png">


tmpParams ends up like this: 
<img width="1328" alt="image" src="https://user-images.githubusercontent.com/29113963/163661762-48cf436f-0e1b-49af-aed4-d4615f4b9f63.png">

Reply:
Also to your point, where the interpolation would only pick up the last value, I also tried it w/ {{path[3]}}, and this is what I got: 

<img width="1021" alt="image" src="https://user-images.githubusercontent.com/29113963/163661896-b29a6fdc-ce9e-4936-b69c-9f6282b70a38.png">

<img width="1277" alt="image" src="https://user-images.githubusercontent.com/29113963/163661921-c9a12992-d5ab-4551-a678-ee2e2fec08b7.png">